### PR TITLE
Plumb through configuration option to disable watch cache

### PR DESCRIPTION
**Pull request by timothysc**
We are seeing anomolies on our load-balanced HA cluster relating to this change, so we would like to plumb through the option to disable.  

In general, I feel we should be doing this more often with new feature additions. 

/cc @wojtek-t

### CODE REVIEW

File: cmd/kube-apiserver/app/server.go [hunk 1]
```diff
@@ -222,6 +223,7 @@ func (s *APIServer) AddFlags(fs *pflag.FlagSet) {
 	fs.Var(&s.RuntimeConfig, "runtime-config", "A set of key=value pairs that describe runtime configuration that may be passed to the apiserver. api/<version> key can be used to turn on/off specific api versions. api/all and api/legacy are special keys to control all and legacy api versions respectively.")
 	fs.StringVar(&s.ClusterName, "cluster-name", s.ClusterName, "The instance prefix for the cluster")
 	fs.BoolVar(&s.EnableProfiling, "profiling", true, "Enable profiling via web interface host:port/debug/pprof/")
+	fs.BoolVar(&s.EnableWatchCache, "watch-cache", true, "Enable watch caching in the apiserver")
```
Comment (wojtek-t) [inline cmd/kube-apiserver/app/server.go]: Can you please add a TODO to enable cache in integration tests (in those tests apiservers are created their own configs, and default of the default config will be false.

**Comment by wojtek-t** [comment]
One comment - other than that looks good to me.

**Comment by wojtek-t** [comment]
btw - thanks for doing this :)

**Comment by timothysc** [comment]
k, modi'd

**Comment by wojtek-t** [comment]
@timothysc travis failed with the following error:
The command "./hack/verify-flags-underscore.py" exited with 1.

**Comment by timothysc** [comment]
@bprashanth ^ build flake.

**Comment by wojtek-t** [comment]
@k8s-bot test this please

**Comment by wojtek-t** [comment]
LGTM - thanks for the fix!

**Comment by timothysc** [comment]
I blame @thockin for bot spam.

**Comment by quinton-hoole** [comment]
e2e failure was due to a flaky cleanup (see below).  Try again...

@k8s-bot test this again please

```
Ran 86 of 150 Specs in 662.264 seconds
SUCCESS! -- 86 Passed | 0 Failed | 2 Pending | 62 Skipped 
```

```
+++ [0903 14:59:38] Cleaning out local _output directory
rm: cannot remove `/jenkins-master-data/jobs/kubernetes-pull-build-test-e2e-gce/workspace/_output/dockerized/bin/darwin/amd64/e2e.test': Permission denied
...
!!! Error in build/../build/common.sh:272
  'rm -rf "${LOCAL_OUTPUT_ROOT}"' exited with status 1
Call stack:
  1: build/../build/common.sh:272 kube::build::clean_output(...)
  2: build/make-clean.sh:26 main(...)
Exiting with status 1
make: *** [clean] Error 1
Build step 'Execute shell' marked build as failure
```

**Comment by quinton-hoole** [comment]
The following test still seems to be slightly flaky, and failed.

Kubernetes e2e suite.Services should serve a basic endpoint from pods

Retrying until we can deflake it.

**Comment by quinton-hoole** [comment]
@k8s-bot test this again please

**Comment by timothysc** [comment]
So I'm about 99.99% certain there are some flakes in the tests. 
Any recommendations here?

**Comment by wojtek-t** [comment]
Yes - this Jenkins failure is flake again:
Services should serve multiport endpoints from pods

@k8s-bot test this please

**Pull request merged**